### ISSUE TITLE
Is the GuildMemberAdd event broken?

### ISSUE THREAD

**Issue by sp8cemonkey83** [bwmarrin/discordgo#793]
Greetings,

I have a working Discord chat bot I've been developing privately for the last year. Recently, I've decided I wanted to run code when new user's join my Discord server that my bot is on.

In my main() function, I am calling the AddHandler function like I do for the messageCreate event:
`client.AddHandler(guildMemberAdd)`

Additionally, I have a guildMemberAdd function declared:
```
func guildMemberAdd(s *discordgo.Session, gma *discordgo.GuildMemberAdd) {
	log.Println("guildMemberAdd event detected")
        session.ChannelMessageSend(<redacted channel id>, "guildMemberAdd event detected")
}
```

I created a brand new Discord account as a test, created an invite on the server the bot is running on, and no matter how many times my new account joins and leaves the server, my bot doesn't detect the event.

To try and debug this a bit, I created another handler to detect all Discord events via the following:
`client.AddHandler(discordEvent)`

```
func discordEvent(s *discordgo.Session, e *discordgo.Event) {
	log.Println(e.Type)
}
```

With this active, I've see many events while interacting in my channels:
TYPING_START
MESSAGE_CREATE
etc. etc.

But when I join the Discord server with my test account, this function doesn't report any new events.

I am wondering if I am just utilizing the API wrong but atleast from what I can see, it seems the guildMemberAdd event is not reporting.

Any help with this would be greatly appreciated!

**Comment by CarsonHoffman** [comment]
Please refer to the [first entry in the FAQ](https://github.com/bwmarrin/discordgo/wiki/FAQ#gateway-intents) that was linked in the [v0.21 release notes](https://github.com/bwmarrin/discordgo/releases/tag/v0.21.1).

**Comment by sp8cemonkey83** [comment]
@CarsonHoffman Thank you for the quick reply.

This did indeed solve my issue. For those that see this, I resolved this issue by doing the following:

Before my client.Open() call, I declared the following:
`client.Identify.Intents = discordgo.MakeIntent(discordgo.IntentsAll)`

Additionally, in the Discord App Portal entry for my bot, I enabled the following:
![image](https://user-images.githubusercontent.com/10186987/87597741-edd9a500-c6b7-11ea-82da-7c948021c701.png)

After all of this, I recompiled, and I am now detecting the GuildMemberAdd event.

I am now closing this issue. Thanks again for the help!
